### ISSUE TITLE
[Hold on #56386][$500] [Simple AA in NewDot] Reports with expenses on Hold by Approver A are being transferred to Approver B when approvers change

### ISSUE THREAD

**Issue by garrettmknight** [Expensify/App#48177]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** N/A
**Reproducible in staging?:** N/A
**Reproducible in production?:** N/A
**Expensify/Expensify Issue URL:**
**Issue reported by:** @garrettmknight 
**Slack conversation:** https://expensify.slack.com/archives/C06ML6X0W9L/p1724850669484329?thread_ts=1724791781.089579&cid=C06ML6X0W9L

## Action Performed:
1. Create a Control workspace in OldDot
2. Invite Submitter, Approver A, Approver B - all as employees
3. Set Submitter's submitsTo to Approver A
4. As Submitter, submits a report with two expenses to Approver A
5. As Approver A, hold one expense and approve the other.
6. As Admin, change Submitter's submits to Approver B
7. As Approver A, approve the unheld expense.

## Expected Result:
No error, held expense is added to new report awaiting Approver A's approval in the processing state

## Actual Result:
Error so Approver A can't approve just the unheld expense. The held expense and report is now assigned to Approver B. Also, Approver B doesn't have GBR or RBR on the report + can't access in NewDot.

## Workaround:
Not really - Approver B can only see the report by directly linking to it.

## Platforms:
All

## Screenshots/Videos

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021836334441203620774</li>
        <li>Upwork Job ID: 1836334441203620774</li>
        <li>Last Price Increase: 2024-10-03</li>
<li>Automatic offers: </li>
<ul>
<li>allgandalf | Reviewer | 104270672</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @garrettmknight</details>

**Comment by garrettmknight** [comment]
WIP, I'm actually seeing the report is rejected back to submitter after step 5.

**Comment by garrettmknight** [comment]
Will retest tomo once our GBR change gets to prod.

**Comment by garrettmknight** [comment]
Didn't get to test today, on my list for tomo

**Comment by garrettmknight** [comment]
Same - will get to this next week.

**Comment by garrettmknight** [comment]
Ok, retested and the rejected expense is still following an unexpected pattern, but I think I know why:
- When an approver holds an expense and approves the remaining expenses, we create a new report for the held expense.
- When hold was introduced, we only had Instant Submit. Now that we have Delay submission, I think we're just defaulting the new report to whatever the Delay submission setting says.
- That means when an approver holds and expense and approves the others, the new report we're creating is created in a draft state, like any other new report would be created in on the delay submission setting.

This feels unexpected because this is effectively rejecting the expense back to the submitter, requiring they submit again in order to have the expense approved.

I think the fix would be to create the new report in the exact state that the original report was in:
- Processing, not draft so we don't require resubmission.
- Awaiting the current approver's approval - this ensures that if something's held later in an advanced workflow that we don't go all the way back to the initial approver.

**Comment by garrettmknight** [comment]
[Confirming expected behavior here](https://expensify.slack.com/archives/C06ML6X0W9L/p1725895753481039)

**Comment by garrettmknight** [comment]
Expected behavior doesn't really exist so we're defining it.

**Comment by garrettmknight** [comment]
Expected behavior is outlined, and OP is updated. Opening it up.

**Comment by robertjchen** [comment]
Thanks! It looks like there's a minor frontend component in updating the existing logic to doing this optimistically as well, we can see if someone at the agency can help out with this part.

In the meantime, I'll update the logic on the backend.

**Comment by hraheel8519** [comment]
Hey,
Here is the  technical explanation: 

Step 1: Identifying the Root Cause

- Investigate the reported issue thoroughly in the linked GitHub issue.
- Review the existing React Native codebase, looking specifically at how the current platform-specific implementations differ.
- Use debugging tools (such as React Developer Tools and Chrome DevTools) to reproduce the issue on various platforms.
- Check for platform discrepancies or unhandled edge cases in component rendering and state management.

Step 2: Plan for Cross-Platform Consistency

- Given that React Native’s goal is to unify front-end development, I will aim to isolate the platform-specific logic and refactor the code to create shared components.
- Implement Platform module from React Native, which allows platform-specific code to be conditionally included.
- Ensure consistent styling across platforms using StyleSheet.create, which helps maintain platform-specific UI.

Step 3: Code Refactor

- Refactor or create a shared component that resolves the issue and works seamlessly across all platforms (iOS, Android, Web).
- Update state management if needed, ensuring it doesn’t rely on platform-specific behaviors.

**Comment by 1subodhpathak** [comment]
Hi! could you please upload a screen recording

**Comment by hraheel8519** [comment]
Can I send you a proposal on upwork?

**Comment by garrettmknight** [comment]
@hraheel8519 we only accept proposals via GH. You'll need to submit a more detailed proposal following our [contribution guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) for us to consider it.

**Comment by Guccio163** [comment]
Hi @garrettmknight, I'm Wiktor Gut from SWM agency, please assign me this issue per @DylanDylann's [request](https://swmansion.slack.com/archives/C04878MDF34/p1727821684544739)

**Comment by Guccio163** [comment]
Oh yes of course, I meant Dylan Courtney; I just picked it today and am reading into this, but I hope I'll get it done quickly 🚀

**Comment by Guccio163** [comment]
Hi, I'm coming with a question that occurred to me when inviting members to reproduce the issue; I invited 3 members to the new workspace and can see all of them, but they doesn't show up as members of this workspace's main chat:

https://github.com/user-attachments/assets/62d989aa-c2ab-44a0-b9f1-7563e5391407

Each one of the members can see themselves and me as one:

https://github.com/user-attachments/assets/1c6c6027-9ef9-49eb-b1c2-3d613781ad22

Should it work like that? It got me confused because in my common sense all of them should be able to read f.ex. expenses made to this workspace. It isn't a front-end bug though, because `Workspace's room members` are extracted from room's `report.participants` and `Workspace's members` are from `policy.employeeList`, I don't quite get it though. Can you please shed some light on this for me?

**Comment by Guccio163** [comment]
Hi, I'm coming with a second question; when user submits an expense to the workspace it just shows in the workspace's chat:
<img width="1261" alt="submitter" src="https://github.com/user-attachments/assets/64738298-58aa-44ec-8e44-5aecfa9145b3">

And when expense's approver receives it, the second chat is created solely for this expense and it isn't visible in the main chat:
<img width="1260" alt="approver1" src="https://github.com/user-attachments/assets/1eac913f-c725-4848-a8ba-460d4a31f98f">
<img width="1255" alt="approver2" src="https://github.com/user-attachments/assets/e780e311-5930-4290-8f56-901377ab05b6">

Why do these two situations differ? I hope that understanding it better can direct us towards the solution since the bug contains not showing report to the original approver.

**Comment by Guccio163** [comment]
Additionally, I (as an approver) can't hold an expense right away, I have to open an expense first:

https://github.com/user-attachments/assets/141b69f7-9d13-4d86-9662-671d4761b656

Is it by design that approver has to open it first?

**Comment by Beamanator** [comment]
Oof i highly doubt it, that looks like a 🐛  - @robertjchen can you confirm?

**Comment by Guccio163** [comment]
Hi, as I'm experimenting a bit with possible expense submissions, couple of not-so-obvious occurrences happened to me; is [this](https://docs.google.com/document/d/12X-hsNo9BBRcG5NSS1JzyjWD1w-iM0uciQJSD1i5mZc/edit#heading=h.bvpgccsp4gwl) the most recent Advanced Approvals design doc? I'd love to check them by myself and gather all these bugs in one place 🔎🪲 fyi I'll be OOO on Monday but I can't wait to patching them on tuesday

**Comment by Guccio163** [comment]
Hi @allgandalf, could you please share a doc/figma for submitting expenses with me? I'm trying to judge whether what I'm encountering is a bug; I also requested a Advanced Approvals figma too, so hopefully I'll be back to speed when it's granted 🙏

**Comment by Guccio163** [comment]
Yes I am thank you, I'll watch closely on any updates 👀

**Comment by Guccio163** [comment]
Hi @allgandalf, today I pulled new main and stumbled over this wall when trying to add an expense to workspace:
<img width="370" alt="Screenshot 2024-10-08 at 12 35 29" src="https://github.com/user-attachments/assets/0b73bb1d-0473-4f53-b674-7064bf6ebcf9">
Do we have an Expensify-friendly burner card to input as a developer? If no, should I go back to older main to postpone that problem?

**Comment by Guccio163** [comment]
Oh okay, I'm currently using 3 '+' accounts for the purposes of this issue, but I guess I'll have to make couple more once they extend 7-days lifespan. Please add my main email: `wiktor.gut@swmansion.com` to the extended accounts list and thanks for the help 😄

**Comment by mallenexpensify** [comment]
> Successfully updated the trial for wiktor.gut@swmansion.com to 2025-01-07 23:59:59 UTC!

@Guccio163 , you should be set! If anyone needs free trials extended for other emails, drop 'em in here.  (caveat, they can't have a balance due on them)

**Comment by Guccio163** [comment]
Hi @mallenexpensify, I'm just checking; Is there a cooldown time after which trial should extend? I'm still getting the same `Add payment card` wall 🚧

**Comment by Guccio163** [comment]
Submitting expense to a newly created workspace doesn't work either 💔

**Comment by Guccio163** [comment]
Hi @allgandalf, as you saw above in the process of reproduction I stumbled over some new bugs. Recently I performed a series of tests to find some other irregularities around submitting, holding and approving the expenses to gather them into one place and fix one after another. 

In total I found about 7 new possible bugs (9 with those above) and would love to talk them over, maybe @garrettmknight or @robertjchen could join the topic to shed some light. Which way do you think would suit the most? It seems a little bit too much for a GH/Slack comment as they all get mixed-up, so maybe a google docs? I'm open for a video call if it would speed up the process :)

I'm waiting for your suggestion and have a great weekend 👋

**Comment by Guccio163** [comment]
Hi @allgandalf 👋 , I didn't work yesterday, so I just started creating repro steps list for you couple of hours ago, I hope to finish it soon 👀

**Comment by Guccio163** [comment]
Hi, I finished creating reproduction steps for the bugs that I've found. In the next comment you'll find 2 notes that I already had before, but I'm bumping it again here; After that there are 9 possible bugs splitted into 5 scenarios (reproduction steps). I'm writing 'possible' because I didn't find any informations about them in design docs, just noted all behaviour that seemed strange to me, please let me know if any of them is there by design. I hope all of them will be clear for you, I'll attach recordings to illustrate bugs too. 🔎🪲

**Comment by Guccio163** [comment]
### Prerequisites: ###
1: Create 4 accounts - Admin, Submitter, Approver 1 and Approver 2
2: Create Workspace as Admin
3: Add Submitter, Approver A and Approver B as members to created Workspace
Prerequisites
Prerequisites:
4: Add workflow: Submitter -> Approver A
5: Turn off Delayed Submissions - bugs available without Delayed Submissions occur with DS too, it's just easier without necessity to submit report separately.



### ADDITIONAL NOTE 1: ###
1. Go to the workspace's chat as a admin - you'll see only yourself as a member.
2. Go to the workspace's chat as a member - you'll see only yourself and Admin as members.

Why does this happen? Shouldn't all workspace's members be listed as a workspace chat's members?

https://github.com/user-attachments/assets/e0562fa6-3ac4-43ce-9616-4e5aabc369c1



### ADDITIONAL NOTE 2: ###
1. Add expense as a Submitter
2. Submitter sees the expense on the main workspace's chat
3. Approver A doesn't see the expense on the main workspace's chat, only in a separate report

Why does this happen? Shouldn't members see all the reports that matter to them in a workspace's chat?



## REPRODUCTION SCENARIOS ##
Steps below don't include relogging as each role, it is assumed that reproducer has to change account to perform an action as someone else.

## SCENARIO 1 ##
1. Submitter adds Expense_1 to Workspace (merchant f.ex. 'a')
2. Approver A opens the workspace's chat, hovers on expense's report and clicks on three-dot

### [POSSIBLE BUG 1] ###
Expected result:
Options open and include "Hold"

Current result:
"Hold" is not included

https://github.com/user-attachments/assets/d354b696-9f1a-44f3-9546-25529ed3bf97



3. Approver A clicks on report and expense's details show
4. Approver A goes back to workspace's reports overview
5. Step 2 again

Current result:
Options open and include "Hold" this time

6. Submitter adds Expense_2 to Workspace (merchant f.ex. 'b')
7. Submitter opens report containing both expenses (they automatically appended to one report)

### [POSSIBLE BUG 2] ###
Expected result:
No errors found

Current result:
Both of them have merchant marked as 'Duplicate', when they clearly aren't duplicated.

Additional notes:
Is it a problem that someone would post two expenses from the same merchant?


https://github.com/user-attachments/assets/601c4d06-33b7-41bd-b786-0ad3c7e246f7




## SCENARIO 2 ##
1. Submitter adds Expense_1 to Workspace
2. Approver A finds expense in inbox
3. Approver A holds expense
4. Submitter finds held expense in inbox
5. Submitter comments on the held expense/changes description (potentially fixing something that could be cause for hold)

### [POSSIBLE BUG 3] ###
Expected result:
Approver A has this expense in inbox because it changed and potentially requires his reassessment.

Current result:
Approver A don't have this expense in inbox anymore. Approver A can find this expense via Search, but have to look for it themselves.


https://github.com/user-attachments/assets/0925127d-e1f8-446c-b58b-879fcd20ee31



6. Approver A comments once again
7. Submitter comments once again, then logs out
8. Submitter sees Expense_1 report in inbox and clicks on it
9. Submitter can see only part of the comments, report disappears from the list on the left and is replaced by another one with (probably) the same reportID as our report. 
10. Submitter clicks on the newly added report and then gets all the information.

### [POSSIBLE BUG 4] ###
Expected result:
Submitter can see all informations/comments regarding this report right away, report doesn't disappear from the reports/workspaces list on the left

Current result:
Submitter have to click one extra time on the new report to access all informations.


https://github.com/user-attachments/assets/22b31d17-6280-4d7f-a9d1-fdb2ac61e569





## SCENARIO 3 ##
1. Steps 1-3 from SCENARIO 2
2. Admin changes Submitter's approver from Approver A to Approver B
3. Approver A still has held expense report in inbox and can approve
4. Approver B sees report in search but cannot approve
5. Admin changes Submitter's approver from Approver B to Approver A

[POSSIBLE BUG 3] again
+
### [POSSIBLE BUG 5] ###
Expected result:
Approver B doesn't see the Submitter's report both in inbox and search.

Current result:
Approver B sees report in inbox (infinite load) and doesn't see in search.


https://github.com/user-attachments/assets/efb93746-c7ed-48bd-8681-bd2909fef249



6. Approver A approves the expense
 
[POSSIBLE BUG 5] still occurs




Further prerequisites:
Admin changes workspace's delayed submissions to on - Manually.




## SCENARIO 4: Original bug reproduction ##
1. Submitter adds 2 expenses and submits them (as a one report)
2. Approver A hold one of the expenses and approves the other one
3. The report splits into two reports one for each expense. Because new report is created for the held expense and workspace has delayed submissions, the report isn't submitted automatically (theoretically doesn't have approver assigned)
4. Admin changes Submitter's approver from Approver A to Approver B]
5. Approver B can see the unsubmitted report (not in inbox but search), clicks "Submit"

### [POSSIBLE BUG 6] ###
Expected result:
Approver B can submit report or can't but doesn't see that option, no errors occur

Current result:
Approver B can see and click the "Submit" button, but error occurs and report doesn't get submitted


https://github.com/user-attachments/assets/a5e54ec7-dfb1-420a-aa54-4d973c58eab6



7. Submitter can see both reports, but only the held one in inbox, when clicking on it doesn't display the submit button

### [POSSIBLE BUG 7] ###
Expected result:
Submitter sees the report with submit button after first click

Current result:
Another variation of this report is shown in inbox, clicking on the new one opens the right report with submit button this time


https://github.com/user-attachments/assets/1d2a8bbb-03b7-4e02-bfc5-c16f8425b6c2



8. Submitter can submit the expense
9. Approver B has this held expense in inbox
10. Approver B clicks on held expense, clicks on this report

### [POSSIBLE BUG 8] - similar to bug 7
Expected result:
Approver B see the "Approve" button

Current result:
Approver B doesn't see the "Approve" button

11. Approver B finds the held expense in Search
12. Approver B can approve from report opened through Search
13. Approver B goes to inbox
14. Approver B this time can approve from report opened when navigating to inbox (the same as found through Search)




## SCENARIO 5: ##
[Delayed submission on, Submitter -> approver B]

1. Submitter adds Expense_1, and submits right away
2. Admin changes workspace policy to Delayed Submissions off, Submitter -> Approver A
3. Submitter adds Expense_2

### [POSSIBLE BUG 9] ###
Expensed result:
Expense creates a separate report with approver A set (does not append to previous report to avoid approvers conflict)

Current result:
Expense is appended to the previous expense's report with a "Unexpected error submitting this expense. Please try again later." error


https://github.com/user-attachments/assets/331c28f9-307e-4e03-9924-256061559067



Possible cause:
Automatically submitted expenses tend to append to other not-yet-submitted/held expenses so does happen in this scenario, but due to couple of expenses with different approvers in one report the error occurs

**Comment by garrettmknight** [comment]
@Guccio163 Thanks for the _very_ in-depth testing and write up - I'll try to repro all the bugs you found tomorrow and confirm, but in the meantime, can you get started on the originally reported bug solution?

**Comment by mallenexpensify** [comment]
@Guccio163 apologies for the delayed reply.  If you're still running into issues with test accounts can you add a list of new emails the I can extend free trials on?  (they need to not be old accounts that have a balance due cuz I don't have access to remove the balances).  Thx

**Comment by Guccio163** [comment]
@garrettmknight Hi, I already started to dive into the problem to confirm my suspicions; I had to handle another issue today so I'm gonna need a moment more, but I hope to find it black-on-white in code soon (or white-on-black considering my dark mode haha)

@mallenexpensify Thanks, I already handled it with a couple extra accounts, no need for extended trail rn 👍

**Comment by mallenexpensify** [comment]
@Guccio163 , @garrettmknight is out of office for a few days, he'll reply once back

**Comment by Guccio163** [comment]
Thanks for a heads up @mallenexpensify; As per the original bug:

When approving only one of the expenses, the report splits into 2 separate reports - one is approved and one held. If the workspace has Delay Submissions turned on, whenever the report is created it goes into let's say 'Draft' mode in which it is yet to be submitted. Moreover any report doesn't have approver assigned, until it's submitted. In our situation it is just that, that happens - newly created report (as a copy of previously submitted and them held expense) stops at 'Draft' stage - I feel like @garrettmknight had the same conclusions earlier.

Normally reports after `Hold` doesn't return to `Draft` stage, so it shouldn't happen here as well; it would also solve approver change problem 👌. In my opinion there are two solutions:
1. New report is created from the start in `Submitted` mode with the previous' report's Approver assigned.
2. Right after creating new report with 'Delay Submissions' rules we call function like `SubmitReport` on the newly created report.
Both of these solutions should assign original submitter as a submitter of the newly created report.

I think this is a Backend issue, because the approval on Frontend results in single API call: 'ApproveMoneyRequest' with slightly varying payload depending on f.ex. whether the whole report is approved or just a part of it (just like in our case). Changing payload doesn't solve our issue here. Inferring from payload parameters Backend returns to us new report with specific `optimisticHoldReportID`, but could also return it in `Submitted` mode using passed `reportID` to assign the same approver as previously (not necessary the current approver).

Please let me know how do you think we should handle it when you get back, I hope we'll fix it quickly 👍

**Comment by Guccio163** [comment]
Hi @garrettmknight , to help you with handling reported bugs, I'm coming with research about `Duplicate` showing under the "Merchant" field (reported as possible bug 2); In my opinion this is a backend issue as well, because there is no place in newDot frontend that could set this error (as a value in Onyx). Also, the `Duplicate` under the merchant is actually indicator of `duplicatedTransaction` error,  but the rules as per which two transactions are duplicates are probably written on backend and I don't have access. If it could help I'm living two examples of expenses that have this error:

Every time violation looks like that:

```
transactionViolations_["expenseID1"]
{data:{
	duplicates: ["expenseID2"],
	tooltip: "List the potential duplicate transactions and merge, delete, or ignore them."
	}
name: "duplicatedTransaction"
type: "warning"}
```

transactionViolations_8488234353613686712 exists and points to 5740488022566865425
transactionViolations_5740488022566865425 exists and points to 8488234353613686712

Expense 8488234353613686712:
amount: -600
merchant: ing\\xd
created: 2024-10-18
No description or category

Expense 5740488022566865425:
amount: -600
merchant: pol
created: 2024-10-18
No description or category


transactionViolations_4882370489895547880 exists and points to 5506408477745284065
transactionViolations_5506408477745284065 doesn't exist (so 5506408477745284065 isn't marked as duplicate)

Expense 4882370489895547880:
amount: -100
merchant: fed
created: 2024-10-18
No description or category

Expense 5506408477745284065:
amount: -100
merchant: def
created: 2024-10-18
No description or category

I hope that the evaluation of reported issues I handled is going well and I hope to hear from you soon!

**Comment by garrettmknight** [comment]
@robertjchen can you help out with the backend portions of [this](https://github.com/Expensify/App/issues/48177#issuecomment-2422510728)?

**Comment by garrettmknight** [comment]
@allgandalf I'm going to take all the other reported issues out of this thread to be fixed elsewhere if I can repro. Let's just focus on the originally reported bug in this issue. I think @Guccio163 is right in his recent analysis above and we'll need to make some backend changes to fix this.

**Comment by Guccio163** [comment]
Great, I'll stay and watch how this issue goes from here; I'll eagerly help with the rest of the issues if you'll be able to reproduce @garrettmknight, of course as far as they'll be front-end fixes 👋

**Comment by garrettmknight** [comment]
Still on me to test.

**Comment by garrettmknight** [comment]
Same - shuffling expense priorities today, but I'll be testing this and a bunch of other approval flows tomorrow for QA.

**Comment by garrettmknight** [comment]
I started working through a few of the reported issues and reproduced some, just need to split them out and report them. @Guccio163 Are you good to start work on the issue in the OP?

**Comment by Guccio163** [comment]
Hi @garrettmknight, from my research which I summarised [here](https://github.com/Expensify/App/issues/48177#issuecomment-2422510728), it looks like it's mostly backend issue, WDYT?

**Comment by garrettmknight** [comment]
Ah, gotcha, we'll set it `Internal` then and find an engineer.

**Comment by Guccio163** [comment]
Great, I'll wait for some updates when it's finished 👍

**Comment by garrettmknight** [comment]
Just an update from my side - I ran through all the issues reported here and I've been able to repro three of them that I'll add to my advanced approval flows/bugs suite I'm working on. In the meantime, the OP is still correct.

**Comment by garrettmknight** [comment]
Still waiting for an internal volunteer here.

**Comment by garrettmknight** [comment]
Same as above.

**Comment by garrettmknight** [comment]
Same as above

**Comment by garrettmknight** [comment]
Same as above.

**Comment by garrettmknight** [comment]
Same as above

**Comment by garrettmknight** [comment]
Same as above

**Comment by garrettmknight** [comment]
Same!

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by garrettmknight** [comment]
Still awaiting a volunteer - I'll retest soon since nothing should have changed this.

**Comment by garrettmknight** [comment]
Still awaiting a volunteer - tested and it's still happening as expected.

**Comment by sakluger** [comment]
Whoops, self-assigned last week accidentally.

**Comment by dylanexpensify** [comment]
bringing up in today's weekly update!

**Comment by dylanexpensify** [comment]
Same!

**Comment by garrettmknight** [comment]
This should actually get fixed in Hold V2 @dylanexpensify - We can leave open, but it should get knocked out there.

**Comment by garrettmknight** [comment]
@mvtglobally does the expected outcome happen or something else?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week) 

https://github.com/user-attachments/assets/7e39d792-1de7-49e2-8399-0858d74be035

**Comment by garrettmknight** [comment]
Just tested it and reproduced, I think the order of the steps is a little off:

1. Create a Control workspace in OldDot
2. Invite Submitter, Approver A, Approver B - all as employees
3. Set Submitter's submitsTo to Approver A
4. As Submitter, submits a report with two expenses to Approver A
5. As Approver A, hold one expense.
6. As Admin, change Submitter's submits to Approver B
7. As Approver A, approve only the unheld expense.

That's when the report with the held expense goes back to Draft.

**Comment by garrettmknight** [comment]
Still holding on the Hold update: https://github.com/Expensify/App/issues/56386

**Comment by garrettmknight** [comment]
Still holding on the Hold update: https://github.com/Expensify/App/issues/56386

**Comment by garrettmknight** [comment]
Still holding on the Hold update: https://github.com/Expensify/App/issues/56386

**Comment by garrettmknight** [comment]
Still holding on the Hold update: https://github.com/Expensify/App/issues/56386

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week)

**Comment by garrettmknight** [comment]
Still holding on the Hold update: https://github.com/Expensify/App/issues/56386

**Comment by garrettmknight** [comment]
Still holding on the Hold update: https://github.com/Expensify/App/issues/56386

**Comment by garrettmknight** [comment]
Still holding on the Hold update: https://github.com/Expensify/App/issues/56386

**Comment by garrettmknight** [comment]
Still holding on the Hold update: https://github.com/Expensify/App/issues/56386

**Comment by garrettmknight** [comment]
Still holding on the Hold update: https://github.com/Expensify/App/issues/56386

**Comment by garrettmknight** [comment]
Still holding on the Hold update: https://github.com/Expensify/App/issues/56386

**Comment by garrettmknight** [comment]
Still holding on the Hold update: https://github.com/Expensify/App/issues/56386

**Comment by garrettmknight** [comment]
Still holding on the Hold update: https://github.com/Expensify/App/issues/56386
